res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1459 @@
+			    if (value) {
+                if (engine.getValue(group, "loop_enabled")) {
+                    engine.setValue(group, "reloop_toggle", true);
+                    engine.setValue(group, "reloop_toggle", false);
+                } else {
+                    engine.setValue(group, "beatloop_activate", true);
+                    engine.setValue(group, "beatloop_activate", false);
+                }
+            }
+	}
+    });
+
+    this.loopActive = new components.Button({
+        midi: [0x90 + channelOffset, 0x50],
+        key: "loop_enabled",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    this.loopHalve = new components.Button({
+        midi: [0x90 + channelOffset, 0x12],
+        key: "loop_halve",
+    });
+
+    this.loopDouble = new components.Button({
+        midi: [0x90 + channelOffset, 0x13],
+        key: "loop_double",
+    });
+
+    this.loopShiftBackward = new components.Button({
+        midi: [0x90 + channelOffset, 0x61],
+        key: "beatjump_backward",
+    });
+
+    this.loopShiftForward = new components.Button({
+        midi: [0x90 + channelOffset, 0x62],
+        key: "beatjump_forward",
+    });
+
+    this.loopIn = new components.Button({
+        midi: [0x90 + channelOffset, 0x10],
+        key: "loop_in",
+    });
+
+    this.loopOut = new components.Button({
+        midi: [0x90 + channelOffset, 0x11],
+        key: "loop_out",
+    });
+
+    this.slotSelect = new components.Button({
+        midi: [0x90 + channelOffset, 0x4C],
+        key: "quantize",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    this.reloopExit = new components.Button({
+        midi: [0x90 + channelOffset, 0x4D],
+        key: "reloop_toggle",
+    });
+
+    this.censor = new components.Button({
+        midi: [0x90+channelOffset, 0x15],
+        key: "reverseroll"
+    });
+
+    this.reverse = new components.Button({
+        midi: [0x90+channelOffset, 0x38],
+	key: "reverse",
+        type: components.Button.prototype.types.toggle
+    });
+
+    this.slipButton = new components.Button({
+          midi: [0x90 + channelOffset, 0x40],
+        key: "slip_enabled",
+          type: components.Button.prototype.types.toggle
+    });
+
+    this.vinylButton = new components.Button({
+        midi: [0x90+channelOffset, 0x17],
+        type: components.Button.prototype.types.toggle,
+        input: function(channel, control, value, status, group) {
+            var midiBaseAddress = 0x90,
+                channel = 0x17,
+                midiOut = 0x00;
+
+              if (value) {
+                  theDeck.jog.vinylMode = !theDeck.jog.vinylMode;
+                  midi.sendShortMsg(
+                    midiBaseAddress + channelOffset,
+                    channel,
+                    theDeck.jog.vinylMode
+                  );
+            }
+        }
+    });
+
+    theDeck.jog.vinylMode = 1; // Set to vinylMode, and the vinyl button is lit..
```

Comment:
do this while constructing the JogWheelBasic instead. Also please use types properly (`true` instead of `1`), this isn't 1980's C code...